components/ILIAS/Forum/classes/Tables/ForumStatisticsTable.php

```diff
@@ -0,0 +1,240 @@
+                $this->lng->txt('learning_progress')
+            )->withIsSortable(false);
+        }
+        return $columns;
+    }
+
+    protected function getDataRetrieval(): DataRetrieval
+    {
+        $data_retrieval = new class (
+            $this->ui_factory,
+            $this->ui_renderer,
+            $this->forum,
+            $this->obj_properties,
+            $this->has_active_lp,
+            $this->has_general_lp_access,
+            $this->completed,
+            $this->in_progress,
+            $this->failed,
+            $this->actor,
+            $this->lng,
+            $this->has_rbac_or_position_access,
+        ) implements DataRetrieval {
+            private ilLPStatusIcons $icons;
+            private ?array $records = null;
+
+            public function __construct(
+                protected \ILIAS\UI\Factory $ui_factory,
+                protected UIRenderer $ui_renderer,
+                protected ilObjForum $object,
+                protected ilForumProperties $obj_properties,
+                protected bool $has_active_lp,
+                protected bool $has_general_lp_access,
+                protected array $completed,
+                protected array $in_progress,
+                protected array $failed,
+                protected ilObjUser $actor,
+                protected ilLanguage $lng,
+                protected bool $has_rbac_or_position_access,
+            ) {
+                $this->icons = ilLPStatusIcons::getInstance(ilLPStatusIcons::ICON_VARIANT_LONG);
+            }
+
+            private function initRecords(): void
+            {
+                if ($this->records === null) {
+                    $this->records = [];
+                    $data = $this->object->Forum->getUserStatistics($this->obj_properties->isPostActivationEnabled());
+                    $result = [];
+                    $counter = 0;
+                    foreach ($data as $row) {
+                        $this->records[$counter]['usr_id'] = $row['usr_id'];
+                        $this->records[$counter]['ranking'] = $row['num_postings'];
+                        $this->records[$counter]['login'] = $row['login'];
+                        $this->records[$counter]['lastname'] = $row['lastname'];
+                        $this->records[$counter]['firstname'] = $row['firstname'];
+                        if ($this->has_active_lp && $this->has_general_lp_access) {
+                            $this->records[$counter]['progress'] = $this->getProgressStatus($row['usr_id']);
+                        }
+                        ++$counter;
+                    }
+                }
+            }
+
+            private function sortedRecords(array $records, Order $order): array
+            {
+                [$order_field, $order_direction] = $order->join([], fn($ret, $key, $value) => [$key, $value]);
+                usort($records, static fn($a, $b) => $a[$order_field] <=> $b[$order_field]);
```

Comment:
You can use this comparison function for numeric sortations, for string based comparisons please use the implementation I chose for `SAML`, as long as there is no centralized service for table sortations available.